File: stepik-union/src/main/java/org/stepik/plugin/actions/navigation/StepikPreviousStepAction.java
package org.stepik.plugin.actions.navigation;

import com.intellij.icons.AllIcons;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.stepik.core.courseFormat.StudyNode;
import org.stepik.plugin.actions.ActionUtils;

public class StepikPreviousStepAction extends StepikStepNavigationAction {
    private static final String ACTION_ID = "STEPIK.PreviousStepAction";
    private static final String SHORTCUT = "ctrl pressed COMMA";
    private static final String SHORTCUT_TEXT = ActionUtils.getShortcutText(SHORTCUT);
    private static final String TEXT = "Previous Step (" + SHORTCUT_TEXT + ")";
    private static final String DESCRIPTION = "Navigate to the previous step";

    public StepikPreviousStepAction() {
        super(TEXT, DESCRIPTION, AllIcons.Actions.Back);
    }

    @Override
    protected StudyNode getTargetStep(@Nullable final StudyNode sourceStepNode) {
        return StudyNavigator.previousLeaf(sourceStepNode);
    }

    @NotNull
    @Override
    public String getActionId() {
        return ACTION_ID;
    }

    @Nullable
    @Override
    public String[] getShortcuts() {
        return new String[]{SHORTCUT};
    }
}

File: stepik-union/src/main/kotlin/org/stepik/plugin/actions/navigation/StepikPreviousStepAction.kt
package org.stepik.plugin.actions.navigation

import com.intellij.icons.AllIcons
import com.intellij.openapi.project.Project
import org.stepik.core.actions.getShortcutText
import org.stepik.core.actions.navigation.StudyNavigator.previousLeaf
import org.stepik.core.courseFormat.StudyNode

class StepikPreviousStepAction : StepikNavigationAction(TEXT, DESCRIPTION, AllIcons.Actions.Back) {

    override fun getTargetStep(project: Project, currentStepNode: StudyNode?): StudyNode? {
        return previousLeaf(currentStepNode)
    }

    override fun getActionId() = ACTION_ID

    override fun getShortcuts() = arrayOf(SHORTCUT)

    companion object {
        private const val ACTION_ID = "STEPIK.PreviousStepAction"
        private const val SHORTCUT = "ctrl pressed COMMA"
        private val SHORTCUT_TEXT = getShortcutText(SHORTCUT)
        private val TEXT = "Previous Step ($SHORTCUT_TEXT)"
        private const val DESCRIPTION = "Navigate to the previous step"
    }
}
